### PR TITLE
feat: enable uds cli next mode

### PR DESCRIPTION
**Pull request by joelmccoy**
## Description

Enables UDS CLI Next mode.  Also enables the unit tests.

## Related Issue

Relates to CLI-263

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [ ] [Contributor Guide Steps](https://github.com/defenseunicorns/uds-cli/blob/main/CONTRIBUTING.md) followed

### CODE REVIEW

**Review by bradclawsie — DISMISSED** [review]

**Comment by joelmccoy** [comment]
@greptileai review

File: tasks/tests.yaml [hunk 1]
```diff
@@ -3,20 +3,15 @@
 
 tasks:
   - name: unit
-    description: run all the unit tests
+    description: run all package unit tests
     actions:
-      - cmd: go test ./cmd/uds ./internal/legacy/... ./internal/mode ./pkg/legacy/... -failfast -v -timeout 5m
+      - cmd: go test ./cmd/... ./internal/... ./pkg/... -failfast -v -timeout 5m
```
[the remark this answers is not in the document]
Comment (joelmccoy) [reply tasks/tests.yaml:8]: fixed.

File: cmd/uds/main.go [hunk 2]
```diff
@@ -54,7 +55,8 @@ func newRootCommand(selected mode.Mode) (*cobra.Command, error) {
 	case mode.Legacy:
 		return legacycli.NewRootCommand(), nil
 	case mode.Next:
-		return nil, errors.New("NextMode is not available in this UDS CLI release")
+		streams := iostreams.New(os.Stdin, os.Stdout, os.Stderr)
+		return cli.NewRootCommand(streams), nil
```
[the remark this answers is not in the document]
Comment (joelmccoy) [reply cmd/uds/main.go]: good catch... routed this better and added tests to check that they are forwarded

[the remark this answers is not in the document]
Comment (joelmccoy) [reply cmd/uds/main.go]: good catch... routed this better and added tests to check that they are forwarded

File: tasks/tests.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (joelmccoy) [reply tasks/tests.yaml:8]: fixed

File: cmd/uds/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (joelmccoy) [reply cmd/uds/main.go]: reattached these

**Comment by joelmccoy** [comment]
@codex review

**Review by bradclawsie — DISMISSED** [review]

File: internal/mode/mode.go [hunk 3]
```diff
@@ -178,6 +178,9 @@ func zarfCommandIndex(args []string) int {
 		if arg == "zarf" || arg == "z" {
 			return i
 		}
+		if arg == "tools" && i+1 < len(args) && args[i+1] == "zarf" {
```
Comment (chance-coleman) [inline internal/mode/mode.go]: related to latest codex comment, if a user puts a normal UDS flag between tools and zarf, the command still works, but Zarf loses its feature flags. Should the argument scanner continue past those UDS flags instead of giving up?

Comment (joelmccoy) [reply internal/mode/mode.go]: Fixed. The Zarf feature scanner now continues through UDS root flags between tools and Zarf, including value-taking flags such as `--log-level debug` or `-l debug`. Regression coverage has been added for both `zarf` and `z` alias paths.

File: internal/cli/zarf/zarf.go [hunk 4]
```diff
@@ -122,10 +119,8 @@ func NewInternalZarfCommand() *cobra.Command {
 		Use:    "zarf",
 		Short:  "Vendored Zarf CLI (internal)",
 		Hidden: true,
-		Run: func(_ *cobra.Command, _ []string) {
-			// Remove "uds" from args, keeping "zarf" and everything after.
-			// os.Args: ["/path/to/uds", "zarf", ...] → ["zarf", ...]
-			os.Args = os.Args[1:]
+		Run: func(_ *cobra.Command, args []string) {
+			os.Args = append([]string{"zarf"}, args...)
```
[the remark this answers is not in the document]
Comment (joelmccoy) [reply internal/cli/zarf/zarf.go:123]: I couldn't reproduce this locally using `CLI_FEATURES=NextMode=true ./build/uds --log-level debug zarf tools kubectl version --client`. The vendored `kubectl` resolves correctly. The related feature-forwarding edge case under `tools ... zarf` has been fixed with tests.

File: internal/mode/mode.go (hunk 3 above)
Comment (chance-coleman) [reply internal/mode/mode.go]: Looking at your response to the codex comment, and im kind of confused by the behavior. I'm seeing the same thing as you with this command: `CLI_FEATURES=NextMode=true ./build/uds --log-level debug zarf tools kubectl version --client` however when i run this command: `CLI_FEATURES=NextMode=true ./build/uds --log-level debug zarf tools kubectl get --help` it panics and outputs this:
```
unable to redefine 'l' shorthand in "get" flagset:
it's already used for "selector" flag
```

My original comment is fixed. I might just be doing something wrong here.

Comment (joelmccoy) [reply internal/mode/mode.go]: ahh goood catch. i update the logic to detect if using zarf tools and pre strip those flags... i added tests to cover these scenarios

**Comment by joelmccoy** [comment]
@codex review

File: internal/mode/mode.go [hunk 5]
```diff
@@ -32,7 +33,7 @@ type FeatureSet map[string]bool
 var processArgs = append([]string(nil), os.Args[1:]...)
 
 func init() {
-	os.Args = append(append([]string(nil), os.Args[:1]...), stripBootstrapArgs(processArgs)...)
+	os.Args = append(append([]string(nil), os.Args[:1]...), prepareBootstrapArgs(processArgs)...)
```
[the remark this answers is not in the document]
Comment (joelmccoy) [reply internal/mode/mode.go:36]: I verified this with built binaries using `GODEBUG=inittrace=1`, including a release-style build with `ActionsCommandZarfPrefix=zarf`. In both runs, `internal/mode` initialized before `zarf/src/cmd`, so the bootstrap argument cleanup occurred before Zarf built its command tree. I also confirmed that `CLI_FEATURES=NextMode=true ./uds --log-level debug zarf tools kubectl get --help` exits successfully.

don't think this is an issue we'd run into.

**Review by slaskawi — APPROVED** [review]

**Review by chance-coleman — APPROVED** [review]
